mismi-s3/src/Mismi/S3/Data.hs

```diff
@@ -80,6 +82,15 @@ renderErrorType e = case e of
   CopyError ->
     "copy"
 
+
+data UploadResult =
+    UploadOk
+  | UploadError UploadError
+
+data UploadError =
+    UploadSourceMissing FilePath
+  | UploadDestinationExists Address
+
```

Comment:
I like this approach. Btw potentially is there any other error which can happen during the upload which we would want to explicitly specify here instead of passing it as exception?
Related point: my bad I think it is a bug I introduced: [this probably should not swallow exception](https://github.com/ambiata/mismi/pull/187/files#diff-1edb1b9e0d390245ba2b012dc71d0c42L201) I should have used `onException` here instead.